CMakeLists.txt

```diff
@@ -43,6 +43,13 @@ option(BUILD_DAEMON "Build bitcoind executable." ON)
 option(BUILD_CLI "Build bitcoin-cli executable." ON)
 option(BUILD_TX "Build bitcoin-tx executable." ON)
 option(BUILD_UTIL "Build bitcoin-util executable." ON)
+# We do not use CMake's BUILD_SHARED_LIBS option.
```

Comment:
Would be good to explain why not.

Reply:
Using [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) makes sense when the project builds shared or static libraries exclusively.

The current branch implementation allows the user to build a shared and a static library simultaneously.

However, I consider this topic as open for discussions because, for example, in the secp256k1 repo, the opposite approach has been suggested and implemented by @theuni.

Reply:
If this remains as it is, then maybe answer the question in the comments, in case other people wonder too:

```suggestion
# We do not use CMake's BUILD_SHARED_LIBS option because using BUILD_SHARED_LIBS makes sense when the project builds shared or static libraries exclusively.
```

Reply:
> Using [`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) makes sense when the project builds shared or static libraries exclusively.
> 
> The current branch implementation allows the user to build a shared and a static library simultaneously.
> 
> However, I consider this topic as open for discussions because, for example, in the secp256k1 repo, the opposite approach has been suggested and implemented by @theuni.

Yeah, I haven't had a chance to go over this in detail yet, but concept NACK on that. libsecp256k1 ended up being much simpler after we switched it to static OR shared. I don't think we should fight CMake's conventions on this one. It really wants to build one type of lib at a time, and I don't think it's a good idea for us to add a bunch of logic to get our old autotools behavior back.

Plus, It's very easy with cmake to make multiple build dirs, one for each library type.

Reply:
This commit will be a part of the following PR and removed from this one.